src/opm/parser/eclipse/EclipseState/EclipseConfig.cpp

```diff
@@ -50,14 +50,17 @@ namespace Opm {
 
         return result;
     }
+    
+    InitConfig& EclipseConfig::init() {
+        return const_cast<InitConfig &>(this->m_initConfig);
```

Comment:
You don't need the `const_cast` here.  `m_initconfig` is a non-const data member in `EclipseConfig` so
```C++
return this->m_initconfig;
```
will return a reference to a mutable `InitConfig` instance.